### PR TITLE
nrt_utils support defaultCredentialsProvider

### PR DESCRIPTION
**Pull request by waziqi89**
RP-9895

Support default credential providers for nrt_utils.
This allows us to use instance profile for AmazonS3.

manual test from dev: https://fluffy.yelpcorp.com/i/mfBgzt5W6T7s6KTNV1n38SpSDRg0gkcm.html

edge case:
When a user want to use default cred-file-path with a non-default profile name:
1. manually put `-c ~/.aws/credentials -p profile-name`
2. provide profile name as an env: AWS_PROFILE=profile-name

### CODE REVIEW

**Review by vim345 — APPROVED** [review]

**Comment by waziqi89** [comment]
Hi @aprudhomme! Just to confirm: how do we apply the change the the tron jobs?
merge this PR, and create a new base image? And then the tron job will automatically pick up the latest utils?

File: src/main/java/com/yelp/nrtsearch/tools/nrt_utils/incremental/DeleteIncrementalSnapshotsCommand.java [hunk 1]
```diff
@@ -69,12 +69,13 @@ public class DeleteIncrementalSnapshotsCommand implements Callable<Integer> {
 
   @CommandLine.Option(
       names = {"-c", "--credsFile"},
-      description = "File holding AWS credentials, uses default locations if not set")
+      description =
+          "File holding AWS credentials, uses default locations if not set; Will use DefaultCredentialProvider if this is unset.")
```
Comment (fragosoluana) [inline src/main/java/com/yelp/nrtsearch/tools/nrt_utils/incremental/DeleteIncrementalSnapshotsCommand.java:73]: Should the description contain only `uses default credential provider chain if this is unset` instead of also including `uses default locations if not set`?

**Review by fragosoluana — APPROVED** [review]
